006-securing-kafka-connect-rest-endpoint.md

```diff
@@ -0,0 +1,146 @@
+# Securing the Kafka Connect REST API endpoint
+
+## Current situation
+
+Currently, instances of Kafka Connect that are deployed by the Strimzi operators are configured with the default REST API endpoint settings. This means that the Kafka Connect REST API endpoint uses HTTP on port 8083 and that the Strimzi KafkaConnect, KafkaConnector and KafkaMirrorMaker2 operators make unsecured REST client calls based on this default configuration.
+
+The default network policies created by the Strimzi operators restrict incoming REST API calls to only allow access from the operator pod. If required, further network policies can be provided to override the default policy and allow wider access.
+
+## Motivation for change
+
+The Kafka Connect REST API endpoint is perhaps the last key endpoint that cannot currently be secured using Strimzi. Many users have requested the ability to secure the Kafka Connect REST API endpoint, for example:
+ - https://github.com/strimzi/strimzi-kafka-operator/issues/3229
+ - https://github.com/strimzi/strimzi-kafka-operator/issues/2161
+ - https://cloud-native.slack.com/archives/CMH3Q3SNP/p1577722052135000
+
+## Proposed changes
+
+This proposal adds a `rest` property to the KafkaConnect spec, allowing users to configure Connect REST API  endpoints that use HTTP or HTTPS. Client authentication is not addressed here, but there is scope for future proposals to address it.
+
+Simple example CR:
+
+```
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+spec:
+  rest:
+  - tls: {}
+  ...
+```
+
+This CR would generate a cert and key in a secret (for use by the operator clients), create SSL truststores and keystores from the cert and key and set the following properties in the generated connect configuration to enable an HTTPS connection:
+
+```
+listeners: https://:8443
+rest.advertised.listener: https
+rest.advertised.port: 8443
+
+listeners.https.ssl.client.auth: none
+listeners.https.ssl.truststore.location: /tmp/kafka/kafka-connect-rest.truststore.p12
+listeners.https.ssl.truststore.password: ***generated password***
+listeners.https.ssl.truststore.type: PKCS12
+listeners.https.ssl.keystore.location: /tmp/kafka/kafka-connect-rest.keystore.p12
+listeners.https.ssl.keystore.password: ***generated password***
+listeners.https.ssl.keystore.type: PKCS12
+
+```
+
+A more complex example CR:
+
+```
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+spec:
+  rest:
```

Comment:
I think it would be good to explicitly describe the  schema of the `rest`. I'm guessing it has `port` (defaulting to ???) and `tls` properties.

Reply:
I have added a `Proposed schema` section in OpenAPI format. Hope it looks OK.